metron-platform/metron-indexing/src/main/java/org/apache/metron/indexing/dao/HBaseDao.java

```diff
@@ -135,8 +138,9 @@ private Document getDocumentFromResult(Result result) throws IOException {
     Map.Entry<byte[], byte[]> entry= columns.lastEntry();
     Long ts = Bytes.toLong(entry.getKey());
     if(entry.getValue()!= null) {
-      String json = new String(entry.getValue());
-      return new Document(json, Bytes.toString(result.getRow()), null, ts);
+      Map<String, Object> json = JSONUtils.INSTANCE.load(new String(entry.getValue()), new TypeReference<Map<String, Object>>() {
+      });
+      return new Document(json, Bytes.toString(result.getRow()), (String) json.get(SOURCE_TYPE), ts);
```

Comment:
I would prefer to see one of two things happen here. Either we keep the constant in the ES specific classes (which is admittedly less than ideal, but it does limit the pollution of ES knowledge into HBase classes) and populate source type from there (basically moving the loading and source type population there).  Alternatively, we pass in a more general function that can be applied to the fields and configure and handle it appropriately.

I think the second one is probably more general useful to be able to do, but given the state of ES5 upgrade making this particular case obsolete, I'm amenable to doing the first option.

At bare minimum we should replace the '.'s with ':'s only if present.  Even if there's not a Solr implementation, I don't want HBaseDao tied to ES so directly.

@cestella Do you have a preference on implementation?  I know you'd had some comments earlier, but I don't want to put words in your mouth.

Reply:
Option 1 is not possible the way things are now.  This constant is already in an ES specific class in metron-elasticsearch but metron-elasticsearch depends on metron-indexing.  To get access to that constant we would need to add metron-elasticsearch as a dependency to metron-indexing thus creating a circular dependency.

Option 2 would be my least favorite option and I would rather just change the method signature of getAllLatest to include all guid/sensorType relationships.

Since we currently have a PR in review for the ES 5 upgrade that will allow us to just remove this constant, I don't feel like we should spend much time on it.  I would also argue that this was an issue long before this PR so anything more than a simple workaround should be a follow-on.

Reply:
I submitted a PR against this PR with my proposed solution of coupling the HBaseDao with the Elasticsearch specific field name transformation logic.  I propose that we do this:
* Migrate the FieldNameConverter as a field in the REST config
* Make the IndexDao take the FieldNameConverter in the AccessConfig
* Have the HBaseDao use the parameterized field name converter to transform `source.type` to the appropriate field.

Reply:
I will amend my constant briefly in that I don't like the first option there as the major flaw in all of these designs is coupling the ES logic within HBase logic.  I think given this is all going away after ES 5 (IMO), we should probably just use the existing abstractions used in the writers to handle field name transformations here where we need to transform field names.

Reply:
I agree with @cestella, I'd rather use the existing abstraction. Setting a field transformer is a cleaner way of doing things, in my opinion, than requiring the user to know know and input where all the GUIDs are coming from.

Another concern about ES5 is that PR may live for awhile, even after we're fairly confident in it (there's going to be a lot of testing involved there). Casey's PR is a pretty simple way of handling it (set a config, pass it in, use it basically the same way you set it up).

Reply:
That seems generally useful but it's a pretty significant change (several files and several lines of code) plus it adds another setting to our config.  There is no unit test for IndexConfig so it also adds untested code (or would require writing a new test).

I prefer a simple 2-3 line change that we can easily revert when ES 5 upgrade makes it in.  This should be a separate PR in my opinion.  This PR is already very large and complicated.

Reply:
The problem with the "we're going to revert after ES 5" argument is that it is likely not going to be immediate.  Until then we have coupled HBaseDao and this feels like a kludge.  

If you don't like this solution, can I propose a compromise that will sidestep this entirely?  How about changing the hbase key in HBaseDao to pass along the sensor (guid + sensor type).  This has a couple of advantages:
* It requires no parsing of the document as it can retrieve the sensor type from the key
* It does not couple our field name conversion from ES with HBase in any way
* It's a small change

I think this neatly sidesteps the issue.  Thoughts?

Reply:
I like this solution much better.

Reply:
Yeah, let's go with this. It solves several problems nicely